scripts/zones/Western_Altepa_Desert/Zone.lua

```diff
@@ -17,6 +18,12 @@ end
 function onInitialize(zone)
     UpdateNMSpawnPoint(ID.mob.KING_VINEGARROON)
     GetMobByID(ID.mob.KING_VINEGARROON):setRespawnTime(math.random(900, 10800))
+
+    dsp.bmt.updatePeddlestox(dsp.zone.WESTERN_ALTEPA_DESERT)
+end
+
+function onGameDay(zone)
+    dsp.bmt.updatePeddlestox(dsp.zone.WESTERN_ALTEPA_DESERT)
```

Comment:
Just pass along the `zone` parameter the function has already.

Reply:
Resolved in https://github.com/DarkstarProject/darkstar/pull/6095/commits/7e36af9746c302b69aef5cb21bd98209fd7e6e59

Reply:
Thought I'd resolved this in 7e36af9, but `onGameDay()` seems to be getting a nil value for `zone`.

Reply:
I guess that's my fault, should have actually looked at what `OnGameDay` was doing instead of assuming based on the parameters in the PR, it doesn't push anything.

I guess remove the parameter and keep pushing the `dsp.zone` constant.